stock_available_to_promise_release/models/stock_move.py

```diff
@@ -145,3 +174,18 @@ def _run_stock_rule(self):
             pulled_moves = pulled_moves.mapped("move_orig_ids")
 
         return True
+
+    def _release_split(self, remaining_qty):
+        """Split move and create a new picking for it.
+
+        Instead of splitting the move and leave remaining qty into the same picking
+        we move it to a new one so that we can release it later as soon as
+        the qty is available.
+        """
+        # Rely on `printed` flag to make _assign_picking create a new picking.
+        # See `stock.move._assign_picking` and
+        # `stock.move._search_picking_for_assignation`.
+        self.picking_id.printed = True
```

Comment:
Maybe worth to check and do the update only if `printed` is not already True, we can avoid several update queries in case of several moves split in the same picking

Reply:
it's unlikely to happen AFAIS, I add the check anyhow.